### DISCUSSION TITLE
Installing/compiling the veramo source code

### DISCUSSION THREAD

**Discussion by praveensankar** [decentralized-identity/veramo#389] [Q&A]
I am looking for any guide/docs to use the veramo source code. It has so many packages in it. But I am not sure where to start and how to see the working. It would be really helpful if someone can point me out to the right direction. I have already tried the cli npm package. I am looking for guide to go through the codebase.

**Comment by mirceanis** [comment]
Please follow the [local development guide](https://github.com/uport-project/veramo#local-development) in the readme file.
This will help you install the dependencies and build locally.
Then, when you want to use a local build in your project instead of the packages released on npmjs you can use `yarn link`.
For example, **after following the build steps in the readme**, you would link the local build of `@veramo/core` to your project like so:
```bash
### make the local build available to other projects
cd /path/to/veramo/sourcecode
cd ./packages/core
yarn link

### instruct your project to use the local build of @veramo/core
cd /path/to/your/project/root
yarn link @veramo/core
```

The [veramo.io documentation site](http://veramo.io/docs/guides/nodejs) has some guides that can get you started. These guides are the same whether or not you use a local build or the official releases.

We are aware that it may seem overwhelming at first and are doing our best to simplify things as much as possible.
It is very important that you go through the [fundamentals sections](http://veramo.io/docs/fundamentals/introduction) and the [agent section](http://veramo.io/docs/agent/introduction) of the docs if you are not familiar with the concepts.

**Reply by praveensankar** [reply]
thank you. I will check those documentations.
